code/crates/test/framework/src/node.rs

```diff
@@ -311,6 +314,13 @@ where
         self
     }
 
+    pub fn with_consensus_disabled(&mut self) -> &mut Self {
+        self.config_customizer = Some(Arc::new(|config| {
+            config.consensus.enabled = false;
+        }));
```

Comment:
Just a note that if we ever add more config customizations, we'll need to either change `config_customizer` to `Vec<ConfigCustomizer<Config>>` or be careful to compose any additional customization function with the existing one, if any.